R/deployApp.R

```diff
@@ -224,16 +224,17 @@ deployApp <- function(appDir = getwd(),
   # wait 1/10th of a second for any queued output get picked by RStudio
   # before emitting the final status, to ensure it's the last line the user sees
   Sys.sleep(0.10)
-  if (!is.null(response$code) && response$code != 0) {
-    displayStatus(paste0(capitalize(assetTypeName), " deployment failed ",
-                         "with error: ", response$error, "\n"))
-    return(invisible(FALSE))
-  } else {
+  deploymentSucceeded <- is.null(response$code) || response$code == 0
+  if (deploymentSucceeded) {
     displayStatus(paste0(capitalize(assetTypeName), " successfully deployed ",
                          "to ", application$url, "\n"))
+  } else {
+    displayStatus(paste0(capitalize(assetTypeName), " deployment failed ",
+                         "with error: ", response$error, "\n"))
   }
 
-  # save the deployment info for subsequent updates
+  # save the deployment info for subsequent updates; we do this even in the
+  # failure case to make it easy to try again
   saveDeployment(appPath,
```

Comment:
Thanks, that's a good idea! On principle, though, I don't like early exits, so I've mostly left that part alone.